Gems/BarrierInput/Code/Source/BarrierInputKeyboard.cpp

```diff
@@ -0,0 +1,258 @@
+            ThreadSafeQueueRawTextEvent(text);
+        }
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceKeyboardBarrier::OnRawKeyboardKeyUpEvent(uint32_t scanCode,
+                                                             [[maybe_unused]]ModifierMask activeModifiers)
+    {
+        // Queue key events, not text events
+        ThreadSafeQueueRawKeyEvent(scanCode, false);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceKeyboardBarrier::OnRawKeyboardKeyRepeatEvent(uint32_t scanCode,
+                                                                 ModifierMask activeModifiers)
+    {
+        // Don't queue key events, only text events
+        if (char asciiChar = TranslateRawKeyEventToASCIIChar(scanCode, activeModifiers))
+        {
+            const AZStd::string text(1, asciiChar);
+            ThreadSafeQueueRawTextEvent(text);
+        }
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceKeyboardBarrier::ThreadSafeQueueRawKeyEvent(uint32_t scanCode, bool rawKeyState)
+    {
+        // From observation, Barrier scan codes in the:
+        // - Range 0x0-0x7F (0-127) correspond to windows scan codes without the extended bit set
+        // - Range 0x100-0x17F (256-383) correspond to windows scan codes with the extended bit set
+        const InputChannelId* inputChannelId = nullptr;
+        if (scanCode < InputChannelIdByScanCodeTable.size())
+        {
+            inputChannelId = InputChannelIdByScanCodeTable[scanCode];
+        }
+        else if (0 <= (scanCode - 0x100) && scanCode < InputChannelIdByScanCodeWithExtendedPrefixTable.size())
+        {
+            inputChannelId = InputChannelIdByScanCodeWithExtendedPrefixTable[scanCode - 0x100];
+        }
+
+        if (inputChannelId)
+        {
+            AZStd::lock_guard<AZStd::mutex> lock(m_threadAwareRawKeyEventQueuesByIdMutex);
+            m_threadAwareRawKeyEventQueuesById[*inputChannelId].push_back(rawKeyState);
+        }
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceKeyboardBarrier::ThreadSafeQueueRawTextEvent(const AZStd::string& textUTF8)
+    {
+        AZStd::lock_guard<AZStd::mutex> lock(m_threadAwareRawTextEventQueueMutex);
+        m_threadAwareRawTextEventQueue.push_back(textUTF8);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    char InputDeviceKeyboardBarrier::TranslateRawKeyEventToASCIIChar(uint32_t scanCode,
+                                                                     ModifierMask activeModifiers)
+    {
+        // Map ASCII character pairs keyed by their keyboard scan code, assuming an ANSI mechanical
```

Comment:
I have no idea how you managed to make this paragraph both left and right justified!

Reply:
It's a habit unfortunately... I have some issues:)

Just removed the last line of the comment as we're no longer using std::unordered_map/pair